src/components/views/Settings.js

```diff
@@ -2,27 +2,51 @@ import {api} from 'helpers/api';
 import {useHistory} from 'react-router-dom';
 import 'styles/views/Settings.scss';
 import {Container, Button, Typography, Box, Slider} from "@mui/material";
-import {useEffect, useState} from "react";
+import {useEffect, useState, useRef} from "react";
 import Button_Click from "./sounds/Button_Click.mp3";
 
 
 const Settings = () => {
     const history = useHistory();
 
-    const [rounds, setRounds] = useState(null);
+    const [rounds, setRounds] = useState(3);
     const [roundTime, setTime] = useState(null);
     const [topic, setTopic] = useState("");
+    const sliderRoundsValueRef = useRef(rounds);
+    const sliderTimeValueRef = useRef(roundTime);
 
     const accessCode = localStorage.getItem('lobbyAccessCode');
-    const handleRoundsChange = (event) => {
-        setRounds(event.target.value)
-    }
-    const handleTimeChange = (event) => {
-        setTime(event.target.value)
+
+    useEffect(() => {
+        const newValue = sliderRoundsValueRef.current;
+        if (!isNaN(newValue)) {
+          setRounds(newValue);
+          doSave();
```

Comment:
As far as I know the setRounds doesn't instantly update the rounds value. Wouldn't it be smarter to not call doSave(). Instead of doSave have a useEffect function which depends on the variable rounds, roundTime and topic.

Reply:
I changed it to only doSave().